ci/infra/testrunner/tests/test_istio_service_mesh.py

```diff
@@ -0,0 +1,159 @@
+
+    kubectl.utils.runshellcommand(istioctl)
+    kubectl.run_kubectl("-n istio-system wait --for=condition=available deploy/istio-ingressgateway --timeout=3m")
+
+    # Activate service in namespace default
+    kubectl.run_kubectl("label namespace default istio-injection=enabled --overwrite")
+
+    # Create the clusterrolebinding to run everything
+    kubectl.run_kubectl("apply -f - << EOF " + CLUSTERROLEBINDING)
+
+    # Deploy bookinfo application
+    kubectl.run_kubectl("apply -f https://raw.githubusercontent.com/istio/istio/release-1.5/samples/bookinfo/platform/kube/bookinfo.yaml")
+    kubectl.run_kubectl("-n default wait --for=condition=available deploy/productpage-v1 --timeout=3m")
+
+    # Deploy the gateway and destination rules
+    kubectl.run_kubectl("apply -f https://raw.githubusercontent.com/istio/istio/release-1.5/samples/bookinfo/networking/bookinfo-gateway.yaml")
+    kubectl.run_kubectl("apply -f https://raw.githubusercontent.com/istio/istio/release-1.5/samples/bookinfo/networking/destination-rule-all.yaml")
+
+    # Deploy the sleep pod that we use as client
+    kubectl.run_kubectl("apply -f https://raw.githubusercontent.com/istio/istio/release-1.5/samples/sleep/sleep.yaml")
+    kubectl.run_kubectl("-n default wait --for=condition=available deploy/sleep --timeout=3m")
+
+
+def _cleanup(kubectl):
+    kubectl.run_kubectl("delete -f https://raw.githubusercontent.com/istio/istio/release-1.5/samples/bookinfo/platform/kube/bookinfo.yaml")
+    kubectl.run_kubectl("delete -f https://raw.githubusercontent.com/istio/istio/release-1.5/samples/bookinfo/networking/bookinfo-gateway.yaml")
+    kubectl.run_kubectl("delete -f https://raw.githubusercontent.com/istio/istio/release-1.5/samples/bookinfo/networking/destination-rule-all.yaml")
+    kubectl.run_kubectl("delete -f https://raw.githubusercontent.com/istio/istio/release-1.5/samples/bookinfo/networking/virtual-service-reviews-v3.yaml")
+    kubectl.run_kubectl("delete -f https://raw.githubusercontent.com/istio/istio/release-1.5/samples/sleep/sleep.yaml")
+    kubectl.run_kubectl("label namespace default istio-injection-")
+
+
+    istioctl_delete = ("""
+                       istioctl --kubeconfig={config} manifest generate \
+                                --set profile=default \
+                                --set addonComponents.prometheus.enabled=false \
+                                --set hub=registry.suse.de/devel/caasp/4.5/containers/containers/caasp/v4.5 \
+                                --set tag=1.5.4 \
+                                --set values.pilot.image=istio-pilot \
+                                --set values.global.proxy.image=istio-proxyv2 \
+                                --set values.global.proxy_init.image=istio-proxyv2 \
+                                | kubectl --kubeconfig={config} delete -f - || true
+                        """.format(config=kubectl.get_kubeconfig()))
+    kubectl.utils.runshellcommand(istioctl_delete)
```

Comment:
I would do this differently: I would run the `istioctl` command first and capture the output and then run the `kubectl` command, so it would be easier to detect errors.

Reply:
I am not running the kubectl command. I am using kubectl library to access the function runshellcommand so that I can execute the istioctl command

Reply:
Maybe I'm reading this wrong but you are piping the output of istioctl to kubectl
```
                              | kubectl --kubeconfig={config} delete -f - || true
```

Reply:
I won't block the PR because of this, but I think that having command in a pipeline in a subcommand will probably make it harder to debug errors.